core/src/main/java/org/apache/iceberg/util/SortOrderUtil.java

```diff
@@ -37,16 +37,14 @@ private SortOrderUtil() {
   }
 
   public static SortOrder buildSortOrder(Table table) {
-    return buildSortOrder(table.spec(), table.sortOrder());
+    return buildSortOrder(table.schema(), table.spec(), table.sortOrder());
   }
 
-  public static SortOrder buildSortOrder(PartitionSpec spec, SortOrder sortOrder) {
+  public static SortOrder buildSortOrder(Schema schema, PartitionSpec spec, SortOrder sortOrder) {
     if (sortOrder.isUnsorted() && spec.isUnpartitioned()) {
       return SortOrder.unsorted();
     }
 
-    Schema schema = spec.schema();
```

Comment:
It wasn't possible to call this method with `PartitionSpec.unpartitioned()` as the schema in the spec was empty.